lib/optimizely/decision_service.rb

```diff
@@ -51,16 +51,7 @@ def get_variation(experiment_key, user_id, attributes = nil)
       # Returns variation ID where visitor will be bucketed (nil if experiment is inactive or user does not meet audience conditions)
 
       # By default, the bucketing ID should be the user ID
-      bucketing_id = user_id;
-
-      # If the bucketing ID key is defined in attributes, then use that in place of the userID
-      if attributes and attributes[RESERVED_ATTRIBUTE_KEY_BUCKETING_ID].is_a? String 
-        unless attributes[RESERVED_ATTRIBUTE_KEY_BUCKETING_ID].empty?
-          bucketing_id = attributes[RESERVED_ATTRIBUTE_KEY_BUCKETING_ID]
-          @config.logger.log(Logger::DEBUG, "Setting the bucketing ID '#{bucketing_id}'")
-        end
-      end
-
+      bucketing_id = get_bucketing_id_from_attributes(user_id,attributes)
```

Comment:
please rename it to get_bucketing_id, it is obvious that it will get from the user attributes. add space after comma, chk existing code.

Reply:
Done.